interfaces/builtin/gpio-control.go

```diff
@@ -0,0 +1,49 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const gpioControlSummary = `allows control of all GPIOs`
+
+const gpioControlBaseDeclarationSlots = `
+  gpio-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const gpioControlConnectedPlugAppArmor = `
+# Description: Allow gpio control
+
+/sys/class/gpio/{un,}export rw,
+/sys/class/gpio/gpio*/{direction,value,edge} rw,
```

Comment:
This includes gpioN as well as gpiochipN. gpiochipN is different and has different files in its directory. Please adjust to be:
```
/sys/class/gpio/gpio[0-9]*/{direction,value,edge} rw,
```

Reply:
Also, I suspect 'active_low' is also needed (since this is 'control').

Reply:
added